File: Selection_Sort.py
def swap(a, b):
    temp = a
    a = b
    b = temp
    return a, b

def selection_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        min_idx = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_idx]:
                min_idx = j
        arr[i], arr[min_idx] = swap(arr[i], arr[min_idx])
    return arr

def print_array(arr):
    for i in arr:
        print(i, end=" ")
    print()

if __name__ == "__main__":
    arr = [19, 4, 95, 22, 53, 11, 90]
    print("Unsorted Array:")
    print_array(arr)
    
    arr = selection_sort(arr)
    
    print("Sorted array in ascending order:")
    print_array(arr)

File: Selection_Sort.cpp
#include <iostream>

void swap(int &a, int &b) {
    int temp = a;
    a = b;
    b = temp;
}

void selectionSort(int arr[], int n) {
    for (int i = 0; i < n - 1; i++) {
        int min_idx = i;
        for (int j = i + 1; j < n; j++) {
            if (arr[j] < arr[min_idx]) {
                min_idx = j;
            }
        }
        swap(arr[i], arr[min_idx]);
    }
}

void printArray(int arr[], int size) {
    for (int i = 0; i < size; i++)
        std::cout << arr[i] << " ";
    std::cout << std::endl;
}

int main() {
    int arr[] = {19, 4, 95, 22, 53, 11, 90};
    int n = sizeof(arr) / sizeof(arr[0]);

    std::cout << "Unsorted Array:" << std::endl;
    printArray(arr, n);

    selectionSort(arr, n);

    std::cout << "Sorted array in ascending order:" << std::endl;
    printArray(arr, n);

    return 0;
}
